src/Microsoft.AspNet.Mvc.WebApiCompatShim/WebApiCompatShimMvcBuilderExtensions.cs

```diff
@@ -18,9 +18,7 @@ public static IMvcBuilder AddWebApiConventions(this IMvcBuilder builder)
             builder.Services.TryAddEnumerable(
                 ServiceDescriptor.Transient<IConfigureOptions<WebApiCompatShimOptions>, WebApiCompatShimOptionsSetup>());
 
-            // The constructors on DefaultContentNegotiator aren't DI friendly, so just
-            // new it up.
-            builder.Services.TryAdd(ServiceDescriptor.Instance<IContentNegotiator>(new DefaultContentNegotiator()));
+            builder.Services.TryAddSingleton<IContentNegotiator, DefaultContentNegotiator>();
```

Comment:
Could just as easily be transient. But that would only create more `DefaultContentNegotiator` instances without changing behaviour (that I've seen). `DefaultContentNegotiator` is not stateful.